runtime/src/proofs.rs

```diff
@@ -0,0 +1,284 @@
+use rstd::vec::Vec;
+
+#[derive(Debug)]
+pub struct Proof {
+    hash: [u8; 32],
+    sorted_hashes: Vec<[u8; 32]>,
+}
+
+/// validates each proof and return true if all the proofs are valid
+/// else returns false
+pub fn validate_proofs(doc_root: [u8; 32], proofs: &Vec<Proof>) -> bool {
+    if proofs.len() < 1 || doc_root.len() < 1 {
+        return false;
+    }
+
+    let mut matches = Vec::new();
+    matches.push(doc_root);
+    return proofs
+        .iter()
+        .map(|proof| {
+            validate_proof(
+                &mut matches,
+                proof.hash.clone(),
```

Comment:
If it doesn't implemeent copy trait, it will be moved. Hence we clone it. But for fixed size array, we do not need to since it implements the copy trait